server/src/main/java/org/apache/seata/server/cluster/raft/manager/RaftControllerStateMachine.java

```diff
@@ -0,0 +1,738 @@
+        boolean leader = isLeader();
+        this.leaderTerm.set(term);
+        LOGGER.info("controllerGroup: {}, onLeaderStart: term={}.", group, term);
+        this.currentTerm.set(term);
+        syncControllerMetadata();
+
+        if (!leader) {
+            CompletableFuture.runAsync(() -> {
+                LOGGER.info("controller became leader, managing {} raft groups", allGroupsMetadata.size());
+                SeataClusterContext.bindGroup(group);
+                try {
+                    // Reload controller metadata and managed groups
+                    reloadControllerState();
+
+                    // Then, initialize TXG groups based on configuration
+                    initializeTxgGroups();
+                } finally {
+                    SeataClusterContext.unbindGroup();
+                }
+            });
+            Configuration conf = RouteTable.getInstance().getConfiguration(group);
+            // A member change might trigger a leader re-election. At this point, it's necessary to filter out
+            // non-existent members and synchronize again.
+            changeControllerPeers(conf);
+        }
+    }
+
+    private void initializeTxgGroups() {
+        try {
+            // 1. Read the TXG configuration
+            String txgGroupsConfig = ConfigurationFactory.getInstance()
+                    .getConfig(SERVER_RAFT_CONTROLLER_GROUP);
+
+            if (StringUtils.isBlank(txgGroupsConfig)) {
+                LOGGER.warn("No TXG groups configured in {}, skipping TXG initialization",
+                        SERVER_RAFT_CONTROLLER_GROUP);
+                return;
+            }
+
+            String[] groupNames = txgGroupsConfig.split(",");
+
+            // 2. Get current controller cluster members
+            Configuration controllerConf = RouteTable.getInstance().getConfiguration(group);
+            if (controllerConf == null) {
+                LOGGER.error("Controller configuration not found for group: {}", group);
+                return;
+            }
+
+            List<PeerId> controllerMembers = controllerConf.getPeers();
+
+            // 3. Validate requirements
+            if (!validateTxgRequirements(groupNames, controllerMembers)) {
+                return; // Validation failed, early return
+            }
+
+            // 4. Proceed with initialization...
+            LOGGER.info("Starting TXG group initialization for: {}", Arrays.toString(groupNames));
+
+            Map<String, List<String>> txgGroupAssignments = createTxgGroupAssignments(
+                    groupNames, controllerMembers);
+
+            submitTxgGroupAssignments(txgGroupAssignments);
+
+        } catch (Exception e) {
+            LOGGER.error("Failed to initialize TXG groups", e);
+        }
+    }
+
+    private boolean validateTxgRequirements(String[] groupNames, List<PeerId> controllerMembers) {
+        // Validate exactly 3 groups
+        if (groupNames.length != 3) {
```

Comment:
I believe a null check is sufficient, as the user might be upgrading from an older version with a single group. In this case, they would only have one group initially, and additional groups can be added later through the API.